qiskit_nature_pyscf/pyscf_ground_state_solver.py

```diff
@@ -0,0 +1,201 @@
+        two_body: np.ndarray
+
+        if restricted_spin:
+            one_body = np.asarray(problem.hamiltonian.electronic_integrals.alpha["+-"])
+            two_body = np.asarray(
+                to_chemist_ordering(
+                    problem.hamiltonian.electronic_integrals.alpha["++--"],
+                )
+            )
+        else:
+            one_body = np.asarray(
+                [
+                    problem.hamiltonian.electronic_integrals.alpha["+-"],
+                    problem.hamiltonian.electronic_integrals.beta["+-"],
+                ]
+            )
+            index_order = find_index_order(problem.hamiltonian.electronic_integrals.alpha["++--"])
+            two_body = np.asarray(
+                [
+                    to_chemist_ordering(
+                        problem.hamiltonian.electronic_integrals.alpha["++--"],
+                        index_order=index_order,
+                    ),
+                    to_chemist_ordering(
+                        problem.hamiltonian.electronic_integrals.alpha_beta["++--"],
+                        index_order=index_order,
+                    ),
+                    to_chemist_ordering(
+                        problem.hamiltonian.electronic_integrals.beta["++--"],
+                        index_order=index_order,
+                    ),
+                ]
+            )
+
+        energy, ci_vec = self.solver.kernel(
+            h1e=one_body,
+            eri=two_body,
+            norb=problem.num_spatial_orbitals,
+            nelec=problem.num_particles,
+        )
+
+        spin_square, _ = fci.spin_square(
+            ci_vec, norb=problem.num_spatial_orbitals, nelec=problem.num_particles
+        )
+
+        density: ElectronicDensity
+        if restricted_spin:
+            raw_density = self.solver.make_rdm1(
+                ci_vec, norb=problem.num_spatial_orbitals, nelec=problem.num_particles
+            )
+            density = ElectronicDensity.from_raw_integrals(raw_density)
+        else:
+            raw_density = self.solver.make_rdm1s(
+                ci_vec, norb=problem.num_spatial_orbitals, nelec=problem.num_particles
+            )
+            density = ElectronicDensity.from_raw_integrals(raw_density[0], h1_b=raw_density[1])
+
+        result = ElectronicStructureResult()
+        result.computed_energies = np.asarray([energy])
+        result.hartree_fock_energy = problem.reference_energy
+        result.extracted_transformer_energies = dict(problem.hamiltonian.constants.items())
+        result.nuclear_repulsion_energy = result.extracted_transformer_energies.pop(
+            "nuclear_repulsion_energy", None
+        )
+        result.num_particles = [sum(problem.num_particles)]
+        result.magnetization = [0.5 * (problem.num_alpha - problem.num_beta)]
+        result.total_angular_momentum = [spin_square]
+        result.electronic_density = density
+
+        return result
+
+    get_qubit_operators = None
```

Comment:
Would it be better to have the method overridden here and simply have it raise an error, in the spirit of the comment below, in case it were used. Would be easier to debug for an end user I feel.